src/main/java/net/imagej/ops/deconvolve/RichardsonLucyUpdate.java

```diff
@@ -52,44 +52,41 @@
  * @param <K>
  * @param <C>
  */
-@Plugin(type = Ops.Deconvolve.RichardsonLucy.class,
+@Plugin(type = Ops.Deconvolve.RichardsonLucyUpdate.class,
 	priority = Priority.HIGH_PRIORITY)
-public class RichardsonLucyImg<I extends RealType<I>, O extends RealType<O>, K extends RealType<K>, C extends ComplexType<C>>
-	extends AbstractFFTFilterImg<I, O, K, C>implements
-	Ops.Deconvolve.RichardsonLucy
+public class RichardsonLucyUpdate<T extends RealType<T>, I extends RandomAccessibleInterval<T>>
+	extends AbstractUnaryComputerOp<I, I> implements
+	Ops.Deconvolve.RichardsonLucyUpdate
 {
 
-	/**
-	 * max number of iterations
-	 */
-	@Parameter
-	private int maxIterations;
+	private BinaryHybridCF<I, I, I> mul = null;
 
-	/**
-	 * indicates whether to use non-circulant edge handling
-	 */
-	@Parameter(required = false)
-	private boolean nonCirculant = false;
+	@Override
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public void initialize() {
 
-	/**
-	 * indicates whether to use acceleration
-	 */
-	@Parameter(required = false)
-	private boolean accelerate = false;
+		// TODO: comment in when problem with initialize is fixed
+		// mul = (BinaryHybridOp) Hybrids.binary(ops(),
+		// IIToIIOutputII.Multiply.class,
+		// RandomAccessibleInterval.class, RandomAccessibleInterval.class,
+		// RandomAccessibleInterval.class);
+	}
 
 	/**
-	 * run RichardsonLucyRAI
+	 * performs update step of the Richardson Lucy Algorithm
 	 */
 	@Override
-	public void runFilter(RandomAccessibleInterval<I> raiExtendedInput,
-		RandomAccessibleInterval<K> raiExtendedKernel, Img<C> fftImg,
-		Img<C> fftKernel, Img<O> output, Interval imgConvolutionInterval)
-	{
+	public void compute1(I correction, I estimate) {
+
+		// TODO: delte these lines when problem in initialization is fixed
+		if (mul == null) {
```

Comment:
@dietzc , @LeonYang5114 , @ctrueden 
``` java
 mul = (BinaryHybridCF)Hybrids.binaryCF(ops(),
		 IIToIIOutputII.Multiply.class,
		 RandomAccessibleInterval.class, RandomAccessibleInterval.class,
		 RandomAccessibleInterval.class);
```
results in 

java.lang.NullPointerException: null
	at net.imglib2.util.Intervals.equalDimensions(Intervals.java:437)
	at net.imagej.ops.math.IIToIIOutputII$Multiply.conforms(IIToIIOutputII.java:237)